src/library/dao/directorydao.cpp

```diff
@@ -144,7 +168,7 @@ DirectoryDAO::RemoveResult DirectoryDAO::removeDirectory(
 QList<RelocatedTrack> DirectoryDAO::relocateDirectory(
         const QString& oldDirectory,
         const QString& newDirectory) const {
-    DEBUG_ASSERT(oldDirectory == mixxx::FileInfo(oldDirectory).location());
+    // DEBUG_ASSERT(oldDirectory == mixxx::FileInfo(oldDirectory).location());
```

Comment:
How can we accomlish the same validation, just without QFileInfo?
IIUC this guarantees that the string and location (string) are identical, if not we can check if it's only the prefix that differs and move on in that case.

Anyone has a smart QString trick handy for this?

Reply:
Actually, do we really need an extensive check here?
It's now the same path returned by `DirectoryDAO::getRootDirStrings()`, so if it was changed on the roundtrip the query will fail.

Reply:
I have omitted the rather complex check (with no actual benefit except debug output) and added a comment instead.
Any ideas to improve the situation are welcome.